File: packages/posthog/ios/Plugin/Classes/Options/StartSessionRecordingOptions.swift
import Foundation
import Capacitor

@objc public class StartSessionRecordingOptions: NSObject {
    private var linkedFlag: Bool?
    private var sampling: Double?

    init(linkedFlag: Bool?, sampling: Double?) {
        self.linkedFlag = linkedFlag
        self.sampling = sampling
    }

    func getLinkedFlag() -> Bool? {
        return linkedFlag
    }

    func getSampling() -> Double? {
        return sampling
    }
}

File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/StartSessionRecordingOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.Nullable;

public class StartSessionRecordingOptions {

    @Nullable
    private Boolean linkedFlag;

    @Nullable
    private Double sampling;

    public StartSessionRecordingOptions(@Nullable Boolean linkedFlag, @Nullable Double sampling) {
        this.linkedFlag = linkedFlag;
        this.sampling = sampling;
    }

    @Nullable
    public Boolean getLinkedFlag() {
        return linkedFlag;
    }

    @Nullable
    public Double getSampling() {
        return sampling;
    }
}
